src/gamestate.rs

```diff
@@ -1,795 +0,0 @@
-use std::collections::HashSet;
-
-use crate::{util::*, BOARD_LENGTH};
-
-#[derive(Clone)]
-struct Selection {
-    starting_tile: Point,
-    current_point: Point,
-    piece: Piece,
-    held_down: bool,
-}
-
-#[derive(Clone)]
-pub struct GameState {
-    pub board: Board,
-    pub last_move: Option<Move>,
-    selected: Option<Selection>,
-    players_turn: Color,
-    prev_game_state: Option<Box<GameState>>,
-    white_pieces: HashSet<Point>,
-    black_pieces: HashSet<Point>,
-    moves_since: i32,
```

Comment:
nit: use a better variable name to convey what this does

Reply:
the name would be like super long like move_since_pawn_move_or_piece_capture